### PR TITLE
Added checks for login fields and some tweaks

### PR DESCRIPTION
**Pull request by Pritom14**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Pritom14** [comment]
@therajanmaurya Sir, I have done as you said, can you please review it

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 1]
```diff
@@ -166,19 +169,17 @@ private boolean isCredentialsValid(final String username, final String password)
             showEmptyInputError(context.getString(R.string.username));
             return false;
         } else if (username.length() < 5) {
-            showMinimumInputLengthNotAchievedError(username,
+            showMinimumInputLengthNotAchievedError("Username",
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:172]: Revert back these changes you can not use hard coded string. I have looked it was written in good manner.

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 2]
```diff
@@ -166,19 +169,17 @@ private boolean isCredentialsValid(final String username, final String password)
...
                     resources.getInteger(R.integer.username_minimum_length));
             return false;
         } else if (username.contains(" ")) {
-            getMvpView().showMessage(context.getString(
-                    R.string.error_validation_cannot_contain_spaces,
-                    username, context.getString(R.string.not_contain_username)));
+            getMvpView().showMessage("Username cannot contain white spaces");
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:176]: Revert back this change, It's unnecessary.

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 3]
```diff
@@ -166,19 +169,17 @@ private boolean isCredentialsValid(final String username, final String password)
...
             return false;
         } else if (password == null || password.trim().isEmpty()) {
-            showEmptyInputError(context.getString(R.string.password));
+            showEmptyInputError("Password cannot be empty");
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:179]: Revert back this change, It's unnecessary.

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 4]
```diff
@@ -166,19 +169,17 @@ private boolean isCredentialsValid(final String username, final String password)
...
             return false;
         } else if (password.length() < 6) {
-            showMinimumInputLengthNotAchievedError(resources.getString(R.string.password),
+            showMinimumInputLengthNotAchievedError("Password",
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:182]: Revert back this change, It's unnecessary.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 5]
```diff
@@ -77,6 +77,7 @@ public void login(final String username, final String password) {
         checkViewAttached();
         getMvpView().showProgress();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:78]: Remove this line, Do not need to show Progressbar before validating the username or password.

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 6]
```diff
@@ -117,6 +118,8 @@ public void onNext(User user) {
                         }
                     })
             );
+        } else {
+            getMvpView().hideProgress();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:122]: No need to hide ProgressBar Unnecessary.

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java (hunk 6 above)
Comment (Pritom14) [reply app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:122]: @therajanmaurya Sir,Can I submit a new PR

**Pull request closed without merging**